diff --git a/ublox_gps/src/gps.cpp b/ublox_gps/src/gps.cpp
index d13c0e9..192ed6d 100644
--- a/ublox_gps/src/gps.cpp
+++ b/ublox_gps/src/gps.cpp
@@ -162,7 +162,8 @@ void Gps::initializeSerial(std::string port, unsigned int baudrate,
   if (config_on_startup_flag_) {
     configured_ = configUart1(baudrate, uart_in, uart_out);
     if(!configured_ || current_baudrate.value() != baudrate) {
-      throw std::runtime_error("Could not configure serial baud rate");
+      // throw std::runtime_error("Could not configure serial baud rate");
+      resetSerial(port_);
     }
   } else {
     configured_ = true;
